x/wasm/keeper/contract.go

```diff
@@ -415,6 +416,21 @@ func (k Keeper) queryToStore(ctx sdk.Context, contractAddress sdk.AccAddress, ke
 	return prefixStore.Get(key)
 }
 
+func assertQueryDepth(ctx sdk.Context) (uint8, error) {
+	var queryDepth uint8
+	if depth := ctx.Context().Value(types.WasmVMQueryDepthContextKey); depth != nil {
+		queryDepth = depth.(uint8)
+	} else {
+		queryDepth = 1
+	}
+
+	if queryDepth > types.ContractMaxQueryDepth {
```

Comment:
If `ContractMaxQueryDepth` cannot be set to <= 0, this assertion scope looks safe to be moved to line 423

Reply:
btw, why can't we just merge line 452 (calling withContext for increasing queryDepth) into this function?